src/components/widgets-common/chart-threshold/chart-threshold.component.coffee

```diff
@@ -74,20 +75,37 @@ module.component('chartThreshold', {
       }]
       return unless ImpacKpisSvc.validateKpiTargets(params.targets)
       promise = if ctrl.isEditingKpi
-        ImpacKpisSvc.update(getKpi(), params, false)
+        ImpacKpisSvc.update(getKpi(), params, false).then(
+          (kpi)->
+            angular.extend(getKpi(), kpi)
+        )
       else
         # TODO: improve the way the hist_params are applied onto widget kpis
         if ctrl.widget.metadata && (widgetHistParams = ctrl.widget.metadata.hist_parameters)
           params.metadata.hist_parameters = widgetHistParams
         else
           params.metadata.hist_parameters = ImpacUtilities.yearDates()
         params.widget_id = ctrl.widget.id
-        ImpacKpisSvc.create('impac', ctrl.kpi.endpoint, ctrl.kpi.watchables[0], params)
+        ImpacKpisSvc.create(
+          ctrl.widget.metadata.bolt_path, ctrl.kpi.endpoint, ctrl.kpi.watchables[0], params
+        ).then(
+          (kpi)->
+            ctrl.widget.kpis.push(kpi)
+            kpi
+        )
       promise.then(
         (kpi)->
-          ctrl.widget.kpis.push(kpi)
-          ctrl.onComplete($event: { kpi: kpi }) if _.isFunction(ctrl.onComplete)
-        (err)->
+          ImpacKpisSvc.show(kpi).then(
+            (response)->
+              dataKey = ImpacKpisSvc.getApiV2KpiDataKey(kpi)
+              angular.extend(kpi, response.data[dataKey])
+            ()->
+              toastr.error('An error occuring while trying to fetch the updated threshold KPI')
```

Comment:
I think you're using the toasters a bit too much: when I add a KPI, a see a line is being drawn, I don't need to be notified. When an error occurs, the "updated threshold KPI" means something to a developer, but I don't think it will make sense to the user...